From 6f3ebdbca5200767c908d0899bb0a142a3667dfb Mon Sep 17 00:00:00 2001
In-Reply-To: <20170608234100.188529-8-bmwill@google.com>
References: <20170608234100.188529-8-bmwill@google.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Fri, 9 Jun 2017 08:39:53 +0000
Subject: [PATCH 1/2] git-compat-util: add a freez() wrapper around free(x); x
 = NULL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a freez() wrapper for the common pattern of freeing a pointer and
assigning NULL to it right afterwards.

The implementation is the same as the (currently unused) XDL_PTRFREE
macro in xdiff/xmacros.h added in commit 3443546f6e ("Use a *real*
built-in diff generator", 2006-03-24).

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 git-compat-util.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/git-compat-util.h b/git-compat-util.h
index 4b7dcf21ad..21e3710755 100644
--- a/git-compat-util.h
+++ b/git-compat-util.h
@@ -787,6 +787,7 @@ extern char *xstrdup(const char *str);
 extern void *xmalloc(size_t size);
 extern void *xmallocz(size_t size);
 extern void *xmallocz_gently(size_t size);
+#define freez(p) do { if (p) { free(p); (p) = NULL; } } while (0)
 extern void *xmemdupz(const void *data, size_t len);
 extern char *xstrndup(const char *str, size_t len);
 extern void *xrealloc(void *ptr, size_t size);
-- 
2.13.0.506.g27d5fe0cd

